### ISSUE TITLE
Dark Background and Light Text addon for chrome?

### ISSUE THREAD

**Issue by EastAPOLO** [m-khvoinitsky/dark-background-light-text-extension#99]
Hello,
I was wondering if its possible for you to port this awesome addon to chrome? :D Not sure if its possible or not, but I am wondering if you ever have plans to do so and to put it on the chrome web store,

Thank you,

EastAPOLO

**Comment by m-khvoinitsky** [comment]
Yes, I have such plans. After WebExtensions port will be ready, I will work on chrome support (current WebExtensions work-in-progress version already more or less works in Chrome) and publishing it to Chrome Web Store.

**Comment by Shickmo** [comment]
Just checking in on this issue since when this gets ported to the chrome store it will also work in vivaldi browser as well. In the current state here are there any basic instructions on using this addon as is in chrome / vivaldi browsers? 

Thanks in advance for all of your awesome work on keeping things updated :) .

**Comment by m-khvoinitsky** [comment]
>In the current state here are there any basic instructions on using this addon as is in chrome / vivaldi browsers?

1.
```bash
git clone https://github.com/m-khvoinitsky/dark-background-light-text-extension.git
cd dark-background-light-text-extension/
git submodule init 
git submodule update
(cd ./lib/webextension-polyfill/; npm install; npm run build)
```
2. Edit `manifest.json`, remove page_action directive
3. Edit `lib/background-script.js`, replace
```javascript
try {
    browser.tabs.insertCSS(sender.tab.id, args);
} catch (e) {
    delete args['cssOrigin'];
    browser.tabs.insertCSS(sender.tab.id, args);
}
```
with
```javascript
delete args['cssOrigin'];
browser.tabs.insertCSS(sender.tab.id, args);
```
Resulting diff should look like this:
```
--- a/lib/background-script.js
+++ b/lib/background-script.js
@@ -90,12 +90,8 @@ browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
                     };
                     if (sender.tab) {
                         // regular tab
-                        try {
-                            browser.tabs.insertCSS(sender.tab.id, args);
-                        } catch (e) {
-                            delete args['cssOrigin'];
-                            browser.tabs.insertCSS(sender.tab.id, args);
-                        }
+                        delete args['cssOrigin'];
+                        browser.tabs.insertCSS(sender.tab.id, args);
                         loaded_stylesheets.push(args);
                     }
                 }
diff --git a/manifest.json b/manifest.json
index f1d6d8f..c36400d 100644
--- a/manifest.json
+++ b/manifest.json
@@ -50,18 +50,6 @@
     "default_title": "Dark Background and Light Text",
     "default_popup": "ui/configure-for-current-tab-panel.html"
   },
-  "page_action": {
-    "browser_style": true,
-    "default_icon": {
-      "16": "icons/icon16.png",
-      "24": "icons/icon24.png",
-      "32": "icons/icon32.png",
-      "48": "icons/icon48.png",
-      "64": "icons/icon64.png"
-    },
-    "default_title": "Dark Background and Light Text preferences for current tab",
-    "default_popup": "ui/configure-for-current-tab-panel.html"
-  },
   "options_ui": {
     "page": "ui/preferences.html",
     "browser_style": true,
```
4. Load this folder as unpacked extension in developer mode.

**Comment by mustafaozhan** [comment]
Thank you so much it works on Chrome !!

I just want to say that i changed my default browser 2 times just because of this extension :)

**Comment by mustafaozhan** [comment]
I can confirm that in some web pages (ex: stackoverflow, medium ) extension not working under chrome. I follow build extension how you said in this topic.

**Comment by Shickmo** [comment]
Hi @m-khvoinitsky ,

I followed the directions you laid out, I am running linux (Solus 3), with Vivaldi version 
1.13.1008.32 (Stable channel) (64-bit) and its not working at the moment. Here is a screenshot of the current warnings I am getting from the extension page and what it looks like when I try to click on the extension widget in the toolbar.

Warnings from extension page:

![image](https://user-images.githubusercontent.com/5192007/33346315-44b312da-d45d-11e7-80c0-657c405bc83b.png)

Picture of clicking on the extension widget in toolbar:

![image](https://user-images.githubusercontent.com/5192007/33346347-67f0a1e0-d45d-11e7-8532-43df441fc703.png)

Any ideas on what I may have done incorrectly or something I could quickly adjust to further test? If not understood, no need to burn cycles on something that you might ultimately fix before chrome store release of the addon anyhow,  I am happy to wait for the chrome store version of the app. For the record your honestly one of my go to addons I can't really live without at this point since the internet is so blindingly white these days ;) . 

Thanks a million for your continued efforts.

--

**Comment by m-khvoinitsky** [comment]
@Shickmo I've tested it on Vivaldi — it works but not on extensions page. Open any regular webpage and try there.

**Comment by Shickmo** [comment]
Hi @m-khvoinitsky ,

I can confirm after performing the steps in the directions you provided that I was able to get the addon installed and confirmed it is mostly working (perhaps I was sleep drunk my first go around). 

One example of a site not working as expected was slashdot.org when set to the default setting. I simply changed it to simple css and everything worked as anticipated but figured I would mention in case someone had not brought this up.

Many thanks for the instructions as this is / was one of the only addon holding me back from going full time with vivaldi. I also might write a little something up on my blog in case there are other die hards of the addon who would like to kick the tires before it hits the chrome store. 

Keep up the amazing work my friend :)

**Comment by m-khvoinitsky** [comment]
Semi-blocker: https://bugs.chromium.org/p/chromium/issues/detail?id=608854

**Comment by mhmt44** [comment]
> I can confirm that in some web pages (ex: stackoverflow, medium ) extension not working under chrome. I follow build extension how you said in this topic.

I can confirm also with chrome 63 and StyleSheet proccessor. Backgrounds are still white (inside of defined color in global settings) It would be perfect if this issue can be solved. I love this extension as a Chrome User (even if it do not exist on store)

**Comment by njohnson7** [comment]
Everything is working great for me, thanks so much! I also deleted the properties in the `manifest` that were causing the errors. Here is the output of my `git diff` if it would help anyone:

```
diff --git a/lib/background-script.js b/lib/background-script.js
index 4e7a3cc..015789f 100644
--- a/lib/background-script.js
+++ b/lib/background-script.js
@@ -92,13 +92,8 @@ browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {
                         runAt: 'document_start',
                     };
                     if (sender.tab) {
-                        // regular tab
-                        try {
-                            browser.tabs.insertCSS(sender.tab.id, args);
-                        } catch (e) {
-                            delete args['cssOrigin'];
-                            browser.tabs.insertCSS(sender.tab.id, args);
-                        }
+                        delete args['cssOrigin'];
+                        browser.tabs.insertCSS(sender.tab.id, args);
                         loaded_stylesheets.push(args);
                     }
                 }
diff --git a/lib/webextension-polyfill b/lib/webextension-polyfill
--- a/lib/webextension-polyfill
+++ b/lib/webextension-polyfill
@@ -1 +1 @@
-Subproject commit 49ce6ef155381585b58213b3487138eac79c5852
+Subproject commit 49ce6ef155381585b58213b3487138eac79c5852-dirty
diff --git a/manifest.json b/manifest.json
index 89e6071..1d8d4b4 100644
--- a/manifest.json
+++ b/manifest.json
@@ -6,11 +6,6 @@
   "icons": {
     "48": "icons/icon48.png"
   },
-  "applications": {
-    "gecko": {
-      "id": "jid1-QoFqdK4qzUfGWQ@jetpack"
-    }
-  },
   "content_scripts": [
     {
       "matches": ["<all_urls>"],
@@ -40,16 +35,19 @@
   },
   "commands": {
     "global_toggle_hotkey": {
-      "suggested_key": {},
+      "suggested_key": {
+        "linux": "Alt+S"
+      },
       "description": "Toggle enabled globally"
     },
     "tab_toggle_hotkey": {
-      "suggested_key": {},
+      "suggested_key": {
+        "linux": "Ctrl+S"
+      },
       "description": "Toggle enabled for current tab"
     }
   },
   "browser_action": {
-    "browser_style": false,
     "default_icon": {
       "16": "icons/icon16.png",
       "24": "icons/icon24.png",
@@ -60,21 +58,8 @@
     "default_title": "Dark Background and Light Text",
     "default_popup": "ui/configure-for-current-tab-panel.html"
   },
-  "page_action": {
-    "browser_style": false,
-    "default_icon": {
-      "16": "icons/icon16.png",
-      "24": "icons/icon24.png",
-      "32": "icons/icon32.png",
-      "48": "icons/icon48.png",
-      "64": "icons/icon64.png"
-    },
-    "default_title": "Dark Background and Light Text preferences for current tab",
-    "default_popup": "ui/configure-for-current-tab-panel.html"
-  },
   "options_ui": {
     "page": "ui/preferences.html",
-    "browser_style": false,
     "open_in_tab": true
   },
   "permissions": [
```

**Comment by ilker4fun** [comment]
> Yes, I have such plans. After WebExtensions port will be ready, I will work on chrome support (current WebExtensions work-in-progress version already more or less works in Chrome) and publishing it to Chrome Web Store.

Is there any news about the development state of chrome support?

**Comment by jezisCoding** [comment]
@njohnson7 
This got me further and even though Chrome still reports errors on the extensions page, it seems to be working. Thanks

**Comment by smaragdus** [comment]
Any news about porting [Dark Background and Light Text](https://github.com/m-khvoinitsky/dark-background-light-text-extension) to Chromium?

**Comment by m-khvoinitsky** [comment]
No, in one of recent versions of the add-on I've started to use one more Firefox-only API (which, by the way, significantly reduced bright flashes) so it became more complicated.

I'll think about it but I can't guarantee anything.

**Comment by smaragdus** [comment]
@m-khvoinitsky 

Thank you for your quick response. For quite some time I have started almost exclusively to use Chromium-based browsers so I cannot use your nice extension on a regular basis.

Anyway, thank you for your good work. I hope that one day you will find time, motivation and suitable APIs to port the extension to Chromium. There are alternatives for Chromium yet I would really like to see your extension ported,

Best regards

**Comment by rublev** [comment]
None of this shit works at all on Mac. Wasted two hours on this.

**Comment by m-khvoinitsky** [comment]
It may be not too obvious, but as I wrote in https://github.com/m-khvoinitsky/dark-background-light-text-extension/issues/99#issuecomment-623451248, currently it's not possible to make the add-on work on Chromium-based browsers anymore. I've edited my comment with outdated instruction.

**Comment by peter-lyons-kehl** [comment]
Hi Mikhail @m-khvoinitsky,

FYI https://issues.chromium.org/issues/40467787 is "fixed". I do __not__ know if it's fixed as needed for us. And, as you've reported, you've been using other Firefox API that doesn't exist (still?) in Chrome/Chromium...

Hoping we could have (even a limited subset of) DBLT on Chrome/Chromium...
